### PR TITLE
feat(breadcrumbs): Fetch parents id & name to build breadcrumbs

### PR DESCRIPTION
**Pull request by trollepierre**
- [x]  propose in desktop as well as in mobile, the display and access to the intermediate folder by a drop-down list when hovering over "...".

- [x]  Use Breadcrumb from MUI

- [x] Public View: to limit this risk of sharing incorrect folder, the breadcrump should always start at the highest point of the tree structure allowed by the token/link. The objective is to make Charlie, as well as any user of the link, understand the real impacts of sharing the URL.

### CODE REVIEW

File: src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx
Comment (trollepierre) [inline src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx:3]: we need to use cozy-ui

File: src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx
Comment (trollepierre) [inline src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx:19]: breadcrumbs link may evolve according to design

File: src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx
Comment (trollepierre) [inline src/drive/web/modules/views/Folder/FolderViewBreadcrumb.jsx:25]: Breadcrumb is used in several places of the apps.

**Review by Ldoppea — APPROVED** [review]
Is the old Breadcrumb still used? Do we expect to replace other occurrences with MUI implementation in the future?

Nit: the commit message does not say that we replaced the breadcrumb component. Maybe you should specify this.

File: src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.jsx
Comment (trollepierre) [inline src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.jsx:37]: @Ldoppea : the Old Breadcrumb is still used in mobile mode

File: src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.spec.jsx
Comment (trollepierre) [inline src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.spec.jsx:10]: @Ldoppea : maybe I should precise here, that in mobile mode, we used the old Breadcrumb. What do you think about it?

Comment (Ldoppea) [reply src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.spec.jsx:10]: Seems like the describe is explicit enough. But I don't really like tests named with just "works". They are not descriptive enough. So your new name improves it a bit, let's keep it.

File: src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.jsx
Comment (trollepierre) [reply src/drive/web/modules/navigation/Breadcrumb/MobileAwareBreadcrumb.jsx]: this line has been deleted. It was not necessary

File: src/drive/web/modules/views/Folder/hooks/useBreadcrumbPath.jsx
Comment (trollepierre) [reply src/drive/web/modules/views/Folder/hooks/useBreadcrumbPath.jsx]: Good catch! 😅 The test was not enough strong by the way, I improved it

**Comment by trollepierre** [comment]
> Is the old Breadcrumb still used? Do we expect to replace other occurrences with MUI implementation in the future?
> Nit: the commit message does not say that we replaced the breadcrumb component. Maybe you should specify this.

The Old Breadcrumb is still used. I will wait to discuss with @Crash-- before taking a decision on this. Then I will update the commit accordingly. :-)

**Comment by trollepierre** [comment]
@Crash-- : can this version demonstrated in the demo be reviewed and merge? or should we arrange the code or wait for Joel approval?

**Comment by trollepierre** [comment]
[📁 DRIVE - BreadCrumb](https://trello.com/c/S9F7SMFM/1785-%F0%9F%93%81-drive-breadcrumb)

**Comment by trollepierre** [comment]
Nouvelle version testable :
https://trollepierre-drivebranch.mycozy.cloud/public?sharecode=KZRNkKm7PYEh#/folder/ff6e94b63cc229b3cfb315d8d786b9b5

Pour tester sur Bender, utiliser `git://github.com/cozy/cozy-drive.git#pierre` comme source

**Comment by trollepierre** [comment]
[📁 DRIVE - Breadcrumb branché sur le point de partage](https://trello.com/c/rkgPknQj/1786-%F0%9F%93%81-drive-breadcrumb-branch%C3%A9-sur-le-point-de-partage)

File: src/drive/web/modules/navigation/Breadcrumb/TODO.md
Comment (Ldoppea) [inline src/drive/web/modules/navigation/Breadcrumb/TODO.md:1]: Seems like this file shouldn't be part of the commit

File: src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl
Comment (Ldoppea) [inline src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl:23]: nit: indentation seems to be incorrect
```suggestion
                min-width 0
```

**Review by Ldoppea — APPROVED** [review]

File: src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl
Comment (Crash--) [inline src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl:45]: nit: We should use semantic variable. But maybe for this one, this is a special case when we don't want to theme?

File: src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl
Comment (Crash--) [inline src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl:66]: Same here

File: src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx
Comment (Crash--) [inline src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx:16]: querySelector can be expensive, right? Should we memoize it ? Or at least store it in order to do not make the call again few lines below? https://github.com/cozy/cozy-drive/pull/2454/files#diff-e47dde8b6893b810a79daecc8827c614ba41ad561395c99789db4027ee222ba2R30

File: src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx
Comment (Crash--) [inline src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx:42]: why don't you pass `setMenuDisplayed` ?

File: src/drive/web/modules/views/Public/index.jsx
Comment (Crash--) [inline src/drive/web/modules/views/Public/index.jsx:32]: import { ROOT_DIR_ID } from 'drive/constants/config'


We need to talk about relative import. Can you add it to our next meeting ? ;)

**Review by Crash-- — APPROVED** [review]

File: src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx
Comment (trollepierre) [reply src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx:16]: good point. That is what was implemented at first. However we faced problems on the public view, when navigating between pages. And we had to come back to this implementation.

So no we don't want to memoize it, because when the second call (line 30) is made, the dropdown trigger could have literally changed from "hidden when less than 3 elements", to "displayed"

The sadness of this code is to have to play with internal component of Material UI... But that was the simplest solution found.

File: src/drive/web/modules/views/Public/index.jsx
Comment (trollepierre) [reply src/drive/web/modules/views/Public/index.jsx:32]: yes, next grooming

File: src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx
Comment (trollepierre) [reply src/drive/web/modules/navigation/Breadcrumb/components/DesktopBreadcrumb/DesktopBreadcrumb.jsx:42]: not necessary: https://reactjs.org/docs/hooks-reference.html#usestate



    Note

    React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.

File: src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl
Comment (trollepierre) [reply src/drive/web/modules/navigation/Breadcrumb/breadcrumb.styl]: i used `primaryTextColor`

**Pull request merged**
